sdk/core/azure-core/src/http/winhttp/win_http_request.hpp

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+// cspell:words PCCERT HCERTSTORE
+
+/**
+ * @file
+ * @brief #Azure::Core::Http::HttpTransport request support classes.
```

Comment:
The reason these types were split out from the public header file is (a) because these are implementation details and are never used by clients of the sdk and (b) including them in the public header requires that I add a TON of WIn32 specific functionality to the public API surface, none of which is necessary.

Splitting them out in a separate PR would be extremely challenging because I would have to add a ton of Win32 specific code to this PR and then immediately remove it. 

It's easier to keep our public API surface clean.